File: checkout.py
#Import the Africa's Talking SDK
import africastalking

#Set up your credentials
username = "sandbox"
api_key = "54a3fb2f30c16e6b1b4df478b5f4395c1705ac253a0b5ecd1a85f9ed296ca18b"

#Initialize the SDK
africastalking.initialize(username, api_key)

#Define the Payment service
check_out = africastalking.Payment

#Set your product name
product_name = "DUKA"

#Set the phone number you want and set it to the international format
phone_number = "+254727545805"

#Set the 3 letter ISO currency code and checkout amount
currency_code = "KES"
amount = 100.0

#You can add in your own metadata which will be resent back to you
#For the final payment notification
metadata = {
    "agentId": "",
    "productId": ""
}

#The provider channel that the payment will be initiated from
provider_channel = "123445"

#Time to send and we'll handle the rest
try:
    res = check_out.mobile_checkout(product_name, phone_number, currency_code, amount, metadata)
    print(res)
except Exception as e:
    print(f"Houston we have a problem {e}")

File: checkout.js
const credentials = {
  apiKey: "62e778b1e5abf6b46a5444e83d424b1337308a5f7b8726408a37217278113fec",
  username: "sandbox"
};

const AfricasTalking = require('africastalking')(credentials);

const payments = AfricasTalking.PAYMENTS;

const initiateMobileCheckout = async () => {
  const options = {
    //Set the product name
    productName: "DUKA",
    phoneNumber: "+254727545805",
    currencyCode: "KES",
    amount: 100,
    metadata: {
      foo: "bar",
      bar: "foo"
    }
  };

  try{
    const result = await payments.mobileCheckout(options);
    console.log(result);
  } catch (err){
    console.log(err);
  }
};

initiateMobileCheckout();
